### PR TITLE
Main menu

### LINKED ISSUE #15: Add main menu to the game
Allow to choose which snakes are competing against eachother.

### LINKED ISSUE #19: Display the time left till the end of the game in score table.

### PR DESCRIPTION
**Pull request by alphabestteam**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+import settings
+from snakes_battle.fruit import FruitKind, Fruit
+
+ai_classes_available = [
+    { "class": RandomSnake, "should_play": False },
+    { "class": SimpleSnake, "should_play": False },
+    { "class": ManualSnake, "should_play": False },
+    { "class": ManualSnakeWASD, "should_play": False }
+    ]
+
```
Comment (eyal1izhaki) [inline main.py]: I would have declare it inside 'main' function and pass it as an argument.

File: main.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            if (menu_action == 'Exit'):
+                return {"action": menu_action, "args": ""}
+            elif (menu_action == 'Play'):
+                return {"action": menu_action, "args": ""}
+
+        elif (snake_picker == True):
+            pass
+
+def run_game(playing_classes):
+    graphics = GameGraphics(ai_classes_available)
```
Comment (eyal1izhaki) [inline main.py:86]: I would have pass the graphics object from main function to this function.

File: snakes_battle/board.py [hunk 3]
```diff
@@ -57,3 +59,12 @@ def get_board_state(self):
...
             }
+
+    def total_game_time(self):
+        return time.time() - self.start_time
+    
+    def game_time_left(self):
+        return round(settings.GAME_TIME_LENGTH - self.total_game_time(), 2)
+    
+    def is_game_timed_out(self):
+        return self.game_time_left() <= 0
```
Comment (eyal1izhaki) [inline snakes_battle/board.py:70]: In my opinion, this code block should be in main module. The timer should start when entering the loop in 'run_game' function.

File: main.py (hunk 2 above)
Comment (alphabestteam) [reply main.py:86]: This is a bug fix.
It started as you say - but for some reason it caused the game after a few runs from the menu to crash.
That fixed it - I guess something in the graphics class is not optimized - but for now that worked well.

**Pull request merged**